DataLayer/Context/GameBoardDbContext.Model.cs

```diff
@@ -9,10 +11,47 @@ protected override void OnModelCreating(ModelBuilder builder)
         {
             base.OnModelCreating(builder);
 
-            builder.Entity<IdentityUser>(
-                entity => entity.ToTable("User"));
+            builder.Entity<ApplicationUser>(
+                entity =>
+                {
+                    entity.Property(e => e.UserName).HasMaxLength(16);
 
-            builder.Entity<IdentityRole>( 
+                    entity.ToTable("User");
+                });
+
+            builder.Entity<Friendship>(
+                entity =>
+                {
+                    entity.HasKey(e => e.Id);
+                    entity.Property(e => e.Id)
+                        .HasMaxLength(32)
+                        .IsRequired()
+                        .ValueGeneratedOnAdd();
+
+                    entity.Property(e => e.FriendshipStatus)
+                        .IsRequired();
+                    entity.Property(e => e.RequestedById)
+                        .IsRequired();
+                    entity.Property(e => e.RequestedToId)
+                        .IsRequired();
+
+                    entity.HasIndex(e => new { e.RequestedById, e.RequestedToId })
+                        .IsUnique();
+
+                    entity.HasOne(e => e.RequestedBy)
+                        .WithMany(u => u.SentRequests)
+                        .HasForeignKey(e => e.RequestedById)
+                        .OnDelete(DeleteBehavior.Restrict);
+                        //Can I set DeleteBehavior to NO ACTION?
+
+                    entity.HasOne(e => e.RequestedTo)
+                        .WithMany(u => u.ReceivedRequests)
+                        .HasForeignKey(e => e.RequestedToId)
+                        .OnDelete(DeleteBehavior.Restrict);
+                        //Can I set DeleteBehavior to NO ACTION?
+                });
```

Comment:
I'd prefer an explicit `entity.ToTable("Friendship")` here, right now the table name is implicitly the same as the entity's name.

EDIT: Actually it's worse, since the table is now called `Friendships`, which is against our naming conventions of singular table names.

Reply:
Fixed